Tests/UnitTests/Diagnostics/FileHandlerTests.swift

```diff
@@ -0,0 +1,301 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  FileHandlerTests.swift
+//
+//  Created by Nacho Soto on 6/16/23.
+
+import Foundation
+import Nimble
+@testable import RevenueCat
+import XCTest
+
+@MainActor
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+class BaseFileHandlerTests: TestCase {
+
+    fileprivate var handler: FileHandler!
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        try AvailabilityChecks.iOS13APIAvailableOrSkipTest()
+
+        self.handler = try Self.createWithTemporaryFile()
+    }
+
+    override func tearDown() async throws {
+        self.handler = nil
+
+        try await super.tearDown()
+    }
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+class FileHandlerTests: BaseFileHandlerTests {
+
+    // MARK: - readFile
+
+    func testReadingEmptyFile() async throws {
+        let data = try await self.handler.readFile()
+        expect(data).to(beEmpty())
+    }
+
+    // MARK: - append
+
+    func testAppendOneLine() async throws {
+        let content = Self.sampleLine()
+
+        await self.handler.append(line: content)
+
+        let data = try await self.handler.readFile()
+        expect(data).to(matchLines(content))
```

Comment:
Should we have a test making sure the file ends with a new line break?

Reply:
I thought it wasn't that important to add an explicit test, as long as the other logic works as expected (appending an existing file, etc). What do you think?

Reply:
Yeah that makes sense 👍